Source/JavaScriptCore/runtime/ObjectConstructorInlines.h

```diff
@@ -83,19 +83,22 @@ ALWAYS_INLINE bool objectAssignFast(JSGlobalObject* globalObject, JSObject* targ
     auto scope = DECLARE_THROW_SCOPE(vm);
     properties.resize(0);
     values.clear();
-    bool canUseFastPath = source->fastForEachPropertyWithSideEffectFreeFunctor(vm, [&](const PropertyTableEntry& entry) -> bool {
-        if (entry.attributes() & PropertyAttribute::DontEnum)
-            return true;
+    bool canUseFastPath = false;
+    if (!source->canHaveExistingOwnIndexedGetterSetterProperties()) {
```

Comment:
How about just returning here early? So then you do not need to change the following.

```
if (source->canHaveExistingOwnIndexedGetterSetterProperties())
    return false;
```